OpenIPSL/Electrical/Renewables/PSSE/PlantController/BaseClasses/BaseREPC.mo

```diff
@@ -0,0 +1,160 @@
+within OpenIPSL.Electrical.Renewables.PSSE.PlantController.BaseClasses;
+partial model BaseREPC
+  "Base class for the plant controller of the WECC generic renewable models"
+
+  import Complex;
+  import Modelica.ComplexMath.arg;
+  import Modelica.ComplexMath.real;
+  import Modelica.ComplexMath.imag;
+  import Modelica.ComplexMath.conj;
+  import Modelica.Blocks.Interfaces.*;
+  parameter OpenIPSL.Types.ApparentPower M_b=SysData.S_b "Machine base power" annotation (Dialog(group="Power flow data"));
+  extends OpenIPSL.Electrical.Essentials.pfComponent(
+    final enabledisplayPF=false,
+    final enablefn=false,
+    final enableV_b=false,
+    final enableangle_0=true,
+    final enablev_0=true,
+    final enableQ_0=true,
+    final enableP_0=true,
+    final enableS_b=true);
+
+  parameter Boolean vcflag=true "Reactive droop (false) or line drop compensation (true)" annotation (Dialog(tab="Control"));
+  parameter Boolean refflag=true "Plant-level reactive power (false) or voltage control (true)" annotation (Dialog(tab="Control"));
+  parameter Boolean fflag=true "Governor response disable (false) or enable (true)" annotation (Dialog(tab="Control"));
+
+  Modelica.Blocks.Interfaces.RealInput Qref "Reactive Power Reference" annotation (Placement(
+        transformation(extent={{-240,100},{-200,140}}), iconTransformation(
+          extent={{-240,100},{-200,140}})));
+  Modelica.Blocks.Interfaces.RealInput Plant_pref "Active Power Reference" annotation (Placement(
+        transformation(extent={{-240,20},{-200,60}}), iconTransformation(
+          extent={{-20,-20},{20,20}},
+        origin={-220,40})));
+  Modelica.Blocks.Interfaces.RealInput Freq "Connection Point Frequency" annotation (Placement(
+        transformation(extent={{-240,-60},{-200,-20}}), iconTransformation(
+          extent={{-240,-60},{-200,-20}})));
+  Modelica.Blocks.Interfaces.RealInput Freq_ref "Plant Controller Frequency Reference"
+                                       annotation (Placement(
+        transformation(extent={{-240,-140},{-200,-100}}),
+        iconTransformation(extent={{-240,-140},{-200,-100}})));
+  Modelica.Blocks.Interfaces.RealOutput Qext(start=q0) "Reactive Power output signal"
+    annotation (Placement(transformation(extent={{200,90},{220,110}})));
+  Modelica.Blocks.Interfaces.RealOutput Pref(start=p0) "Real Power output signal"
+    annotation (Placement(transformation(extent={{200,-110},{220,-90}})));
```

Comment:
p0 and q0 are not parameter expressions but stem from a variable input. This should be changed. All initial value parameters should be modelled as parameters, not real inputs/outputs. This would also clean up the diagrams. The same also applies to the other base classes.